src/services/earn/earn-service.ts

```diff
@@ -1276,3 +1375,28 @@ type GetPositionsParams = BaseGetPositionsParams &
         accounts: ArrayOneOrMore<Address>;
       }
   );
+
+const externalFirewallAbi = [
+  {
+    inputs: [{ internalType: 'bytes4', name: 'selector', type: 'bytes4' }],
+    name: 'getCheckpoint',
+    outputs: [
+      { internalType: 'uint192', name: '', type: 'uint192' },
+      { internalType: 'uint16', name: '', type: 'uint16' },
+      { internalType: 'uint16', name: '', type: 'uint16' },
+      { internalType: 'enum Activation', name: '', type: 'uint8' },
+      { internalType: 'bool', name: '', type: 'bool' },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+] as const;
+
+enum MethodsSelector {
+  EARN_VAULT_COMPANION_WITHDRAW = '0x48316792',
```

Comment:
Can we use viem's `toFunctionSelector`?
https://viem.sh/docs/utilities/toFunctionSelector

So that they are calculated automatically, in case we change the functions in the future